Update glorot_uniform_initializer to match other Initializers

yongtang opened the pull request:
This fix tries to address the issue raised in #19910 where the signature of glorot_uniform_initializer does not match other Initializers (e.g., random_uniform_initializer).

This fix update the glorot_uniform_initializer so that it matches the signature in a similiar way as other
Initializers.

This fix fixes #19910.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update glorot_uniform_initializer to match other Initializers

This fix tries to address the issue raised in 19910 where
the signature of glorot_uniform_initializer does not match
other Initializers (e.g., random_uniform_initializer).

This fix update the glorot_uniform_initializer so that
it matches the signature in a similiar way as other
Initializers.

This fix fixes 19910.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Also convert glorot_normal_initializer to Initializer

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Pylint fix

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

qlzh727 commented:
Nudge @fchollet for review/approval.

tensorflowbutler commented:
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

qlzh727 commented:
Sorry for the late reply. Please resolve the merge conflict, and fix the api compatibility test failure. See  https://source.cloud.google.com/results/invocations/4de5073a-8d47-4424-a7fa-bef0a215d19c/targets/%2F%2Ftensorflow%2Ftools%2Fapi%2Ftests:api_compatibility_test/log

yongtang committed:
Fix merge conflicts

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update API goldens

with
```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

fchollet reviewed (approved):
Thanks for the PR! Changes look good to me.

@qlzh727 we'll need an API owner to validate the API changes in this PR.

qlzh727 commented:
Ack, I will probably patch the PR internally and update the pbtxt.

qlzh727 commented:
Sure. Will cc @martinwicke on the change.

martinwicke commented on tensorflow/python/ops/init_ops.py:
```diff
@@ -1178,16 +1168,45 @@ def glorot_normal_initializer(seed=None, dtype=dtypes.float32):
+convolutional_delta_orthogonal = ConvolutionDeltaOrthogonal
+convolutional_orthogonal_1d = ConvolutionOrthogonal1D
+convolutional_orthogonal_2d = ConvolutionOrthogonal2D
+convolutional_orthogonal_3d = ConvolutionOrthogonal3D
```
While we're here, could we remove this aliases (are they even effective? Do they do anything given that the tf_export decorators determine what's in the API?)?

@annarev do you think these actually do anything?

annarev replied:
Looks like these are used in contrib:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/framework/__init__.py
I suppose they can be imported with "as convolutional_orthogonal_*d" instead.

martinwicke commented:
@tensorflow/api-owners This is an almost-noop in terms of API, I think it's fine.

tensorflowbutler commented:
Nagging Assignee @qlzh727: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
